modules/ROOT/pages/salesforce/salesforce-connector-96.adoc

```diff
@@ -0,0 +1,1211 @@
+
+This procedure provides guidance on using Salesforce to create a consumer key, and explains how to create a connected app in Salesforce. However, note that the steps might differ somewhat in your Salesforce instance.
+
+Prerequisite:
+
+This procedure assumes that you already have a certification file (such as `salesforce-cert.crt`). If not, you can produce one by generating a Java KeyStore and Public Key.
+
+[[create-consumer-key]]
+. Log into Salesforce, and go to *Setup* > *Build* > *Create* > *Apps*.
+. Under the Connected App section, click *New*.
+. Follow these steps to create a new connected app, and enter the following information in the appropriate fields:
++
+* A name for the connected app.
+* The API name.
+* Contact email.
++
+. Under API (Enable OAuth Settings), select Enable OAuth Settings:
++
+* Enter the Callback URL.
+* Select the Use Digital Signatures checkbox.
+* Click Browse (or Choose File), and load your Salesforce certificate (for example, `salesforce-cert.crt`), which contains your public key.
++
+In Studio, you typically store this in the workspace that contains your Mule application.
++
+. Add and Save these OAuth scopes to Selected OAuth Scopes:
++
+Full Access (`full`) and Perform Requests On Your Behalf At Any Time (`refresh_token`, `offline_access`)
++
+. Configure the Authorization Settings for the app:
++
+Click *Manage*. Then under the OAuth Policies section, expand the *Permitted Users* dropdown, and select *Admin Approved Users are Pre-Authorized*. Then Save.
+. Under the Profiles section, click *Manage Profiles*.
+. Select your user profile, and then click *Save*.
+. Go back to the list of Connected Apps: *Build* > *Create* > *Apps*.
+. Under the Connected Apps section, select the connected app you created.
+
+You can see the Consumer Key that you need to provide in your connector's configuration.
+
+== Generate a Keystore File
+
+The *Keystore* is the path to the keystore used to sign data during authentication. Only Java keystore format is allowed.
+
+To generate a keystore file:
+
+. Go to your Mule workspace, and open the command prompt (for Windows) or Terminal (for Mac).
+. Type `keytool -genkeypair -alias salesforce-cert -keyalg RSA -keystore salesforce-cert.jks` and press enter.
+. Enter the following:
++
+** Password for the keystore.
+** Your first name and last name.
+** Your organization unit.
+** Name of your city, state, and the two letters code of your county.
++
+The system generates a java keystore file containing a private/public key pair in your workspace.
++
+. Provide the file path for the *Keystore* in your connector configuration.
++
+Type `keytool -exportcert -alias salesforce-cert -file salesforce-cert.crt -keystore salesforce-cert.jks` and press enter.
++
+The system now exports the public key from the keystore into the workspace. This is the public key that you need to enter in your Salesforce instance.
```

Comment:
"The system"? MuleSoft? Salesforce? OS?

Reply:
What do you mean Salesforce instance? You can't just throw things in the org :)

Reply:
I'm asking for specificity here just because there are so many different things involved, being a little precise may help someone who is new to identity mgmt, SFDC, or MuleSoft?